src/Chapter4.hs

```diff
@@ -489,7 +511,18 @@ Implement the 'Applicative' instance for our 'List' type.
   type.
 -}
 
+append :: List a -> List a -> List a
+append Empty d = d
+append (Cons el l) l2 = Cons el (append l l2) 
```

Comment:
Perfect!

You can one more case to make your solution even more efficient:

```haskell
append xs Empty = xs
```

In that case, you won't need to walk through each element of the first list when the second one is empty and return the list instantly 🚤